### DISCUSSION TITLE
Inconsistent behavior of `pixdim` in `Spacing`

### DISCUSSION THREAD

**Discussion by ChenglongWang** [Project-MONAI/MONAI#4807] [General]
Recently, I noticed that the `pixdim` argument from the `Spacing` transform is inconsistent with other transforms.
For the `pixdim` in `Spacing` transform, `if providing a single number, will use it for the first dimension. items of the pixdim sequence map to the spatial dimensions of input image`. However, for the most of other transforms, such as `Resize`, `SpatialPad`, and `BorderPad`, single number argment will be automatically filled to the image size by using `fall_back_tuple` fn.
I'm wondering why `Spacing` transform is designed this way. Will this inconsistency lead to any improper usage?

**Comment by wyli** [comment]
thanks, I'll create a feature request from this, defaulting to the isotropic resolutions instead of all 1.0 values, though I'm not sure which one is better...

**Reply by ChenglongWang** [reply]
Hi @wyli , thanks for your quick response

I think isotropic resolution is more natural behavior for the single number input to keep consistent with other transforms.
If users need 1.0 value, I think most MONAI users will naturally input (x, 1.0, 1.0).

**Reply by wyli** [reply]
(addressed by https://github.com/Project-MONAI/MONAI/pull/5248)
